### PR TITLE
Add `metric` to list of `--resource-types`

### PR DESCRIPTION
**Pull request by dbeatty10**
The key information is in the source code [here](https://github.com/dbt-labs/dbt-core/blob/30503697f2023c379338da1e58029582499ca877/core/dbt/cli/params.py#L287-L309) and [here](https://github.com/dbt-labs/dbt-core/blob/726800be571004dab47c149a5bd4514186562afc/core/dbt/task/list.py#L20-L32).

## What are you changing in this pull request and why?
- The `metric` resource type was left out of the list, so it was added
    - Also re-order resource types
- It wasn't clear which resources are included in `default` and which are left out, so that was clarified
- It wasn't clear how `default` differs from `all`, but it should be more clear now

## Checklist

- [x] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) and [About versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) so my content adheres to these guidelines.

### CODE REVIEW

File: website/docs/reference/commands/list.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ dbt ls
 See [resource selection syntax](node-selection/syntax) for more information on how to select resources in dbt
 
 **Arguments**:
-- `--resource-type`: This flag limits the "resource types" that dbt will return in the `dbt ls` command. By default, the following resources are included in the results of `dbt ls`: models, snapshots, seeds, tests, and sources.
+- `--resource-type`: This flag limits the "resource types" that dbt will return in the `dbt ls` command. By default, all resources are included in the results of `dbt ls` except for analysis.
```
Comment (mirnawong1) [inline website/docs/reference/commands/list.md]:
```suggestion
- `--resource-type`: This flag restricts the "resource types" returned by dbt in the `dbt ls` command. By default, all resource types are included in the results of `dbt ls` except for the analysis type.
```

**Comment by mirnawong1** [comment]
hey @dbeatty10 - thanks so much for opening this up! i left a small sugg, let me know if you have any questions!

File: website/docs/reference/commands/list.md (hunk 1 above)
Comment (dbeatty10) [reply website/docs/reference/commands/list.md]: Thanks for this suggestion @mirnawong1 !

Two questions:
1. Do you prefer 'dbt ls' over `dbt ls`? If so, I'm curious to hear more why for my edification!
2. How do you feel about this as the final sentence?
    > By default, all resource types are included in the results of `dbt ls` except for the analysis type.

Comment (mirnawong1) [reply website/docs/reference/commands/list.md]: ack! so for sure that was my bad, i prefer `dbt ls` (per style guide too) and i just forgot to use the code apostrophe 🙈 sorry about causing that confusion!

ok i updated your final sentence - what do you think?

Comment (dbeatty10) [reply website/docs/reference/commands/list.md]: Looks great -- committed your suggestion 🙌

**Review by mirnawong1 — APPROVED** [review]
lgtm @dbeatty10 ! ready for you to merge ✨

**Pull request merged**